securedrop/tests/conftest.py

```diff
@@ -172,7 +171,10 @@ def config(
 
         config.SUPPORTED_LOCALES = i18n.get_test_locales()
 
-        yield config
+        # Set this newly-created config as the "global" config
+        with mock.patch.object(sdconfig, "config", config):
```

Comment:
Important change here: this ensures that the application code "sees" the same config as the one created in the fixture (when the code imports the config). 

It's not perfect and more work is needed with the config fixture, but it's a step forward.

Reply:
One change required when running `make test` locally - the SECUREDROP_ENV var needs to be set, as sdconfig uses it to set the database URI to the appropriate value, otherwise alembic tests will fail. (Letting the rest of the tests run now to see if there are any similar issues with other tests).

Reply:
`SECUREDROP_ENV` does not seem to be set here when  `make test` is run locally .  It's used to set the database URI to the appropriate value for the environment,  so without it the alembic tests in particular will fail. (Said failure isn't happening in CI for some reason).

One solution is to add `os.environ["SECUREDROP_ENV"] = "test"` before filesystem setup, but I'm sure there are more elegant ones :)